src/walletConnect/saga.ts

```diff
@@ -361,11 +423,37 @@ function* showActionRequest(request: Web3WalletTypes.EventArguments['session_req
 
   const supportedChains = yield* call(getSupportedChains)
 
+  const networkId = walletConnectChainIdToNetworkId[request.params.chainId]
+  const feeCurrencies = yield* select((state) => feeCurrenciesSelector(state, networkId))
+  let preparedTransactionsResult: PreparedTransactionsResult | undefined = undefined
+  if (
+    method === SupportedActions.eth_signTransaction ||
+    method === SupportedActions.eth_sendTransaction
+  ) {
+    const network = walletConnectChainIdToNetwork[request.params.chainId]
+    const normalizedTx = yield* call(
+      normalizeTransaction,
+      request.params.request.params[0],
+      network
+    )
+    preparedTransactionsResult = yield* call(prepareTransactions, {
+      feeCurrencies,
+      decreasedAmountGasFeeMultiplier: 1,
+      baseTransactions: [normalizedTx],
+    })
+  }
+
   navigate(Screens.WalletConnectRequest, {
     type: WalletConnectRequestType.Action,
     pendingAction: request,
     supportedChains,
     version: 2,
+    hasInsufficientGasFunds: preparedTransactionsResult?.type === 'not-enough-balance-for-gas',
+    feeCurrencies,
```

Comment:
yes.....i kind of accepted this because feeCurrencies is coming directly from redux where we should have no unserializable values anyway (if we fix the redux problem, this would be solved too) but i suppose there's no reason to keep contributing to the problem for this component. we only need to display the symbols so i've updated this to be a string array